### ISSUE TITLE
Allow resolving of serverless variables within the config

### ISSUE THREAD

**Issue by philiiiiiipp** [floydspace/serverless-esbuild#133]
Hey, really awesome initiative, got our build process from ~15 minutes to ~30 seconds!

I noticed one issue with the resolving of the config variables when using e.g. the `${file` import. Those are not resolved during the constructor call yet, e.g.

```yml
esbuild:
  packager: npm
  bundle: true
  define: ${file(./serverless.js):define}
```

will show up as 
```ts
{
  packager: npm,
  bundle: true,
  define: ${file(./serverless.js):define},
}
```

rather than 

```ts
{
  packager: npm,
  bundle: true,
  define: {
    some: 'resolved',
    variables: 123
}
```

I put up a PR #132  moving the config generation outside of the constructor which fixes that behaviour. It does now regenerate the config at every request to `this.buildOptions` but it seemed to me that this might be just as well since there is some deletion going on in the `bundle` step and this is likely super infrequent. If you mind that, I can cache the config file.

**Comment by floydspace** [comment]
Hey, @philiiiiiipp, thank you for using it. I'm really happy that it's useful to people, and even happier when they contribute.

**Comment by floydspace** [comment]
Fixed in #132
